xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/NfTokenIT.java

```diff
@@ -0,0 +1,464 @@
+import org.xrpl.xrpl4j.client.XrplClient;
+import org.xrpl.xrpl4j.model.client.accounts.AccountInfoResult;
+import org.xrpl.xrpl4j.model.client.accounts.AccountNftsResult;
+import org.xrpl.xrpl4j.model.client.accounts.NfTokenObject;
+import org.xrpl.xrpl4j.model.client.nft.NftBuyOffersRequestParams;
+import org.xrpl.xrpl4j.model.client.nft.NftBuyOffersResult;
+import org.xrpl.xrpl4j.model.client.nft.NftSellOffersRequestParams;
+import org.xrpl.xrpl4j.model.client.nft.NftSellOffersResult;
+import org.xrpl.xrpl4j.model.client.transactions.SubmitResult;
+import org.xrpl.xrpl4j.model.flags.Flags;
+import org.xrpl.xrpl4j.model.ledger.NfTokenOfferObject;
+import org.xrpl.xrpl4j.model.transactions.NfTokenAcceptOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenBurn;
+import org.xrpl.xrpl4j.model.transactions.NfTokenCancelOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenCreateOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenId;
+import org.xrpl.xrpl4j.model.transactions.NfTokenMint;
+import org.xrpl.xrpl4j.model.transactions.TransactionResultCodes;
+import org.xrpl.xrpl4j.model.transactions.XrpCurrencyAmount;
+import org.xrpl.xrpl4j.tests.environment.CustomEnvironment;
+import org.xrpl.xrpl4j.tests.environment.XrplEnvironment;
+import org.xrpl.xrpl4j.wallet.Wallet;
+
+public class NfTokenIT extends AbstractIT {
+
+  private static final XrplEnvironment nftDevnetEnvironment = new CustomEnvironment(
+    HttpUrl.parse("http://xls20-sandbox.rippletest.net:51234"),
+    HttpUrl.parse("https://faucet-nft.ripple.com")
+  );
+  private static final XrplClient xrplClient = nftDevnetEnvironment.getXrplClient();
+
+  @Test
+  void mint() throws JsonRpcClientErrorException {
+    Wallet wallet = createRandomAccount();
+
+    AccountInfoResult accountInfoResult = this.scanForResult(
+      () -> this.getValidatedAccountInfo(wallet.classicAddress())
+    );
+    int previousLength = xrplClient.accountNfts(wallet.classicAddress()).accountNfts().size();
+
+    //Nft mint transaction
+    NfTokenMint nfTokenMint = NfTokenMint.builder()
+      .tokenTaxon(UnsignedLong.ONE)
+      .account(wallet.classicAddress())
+      .fee(XrpCurrencyAmount.ofDrops(50))
+      .signingPublicKey(wallet.publicKey())
+      .sequence(accountInfoResult.accountData().sequence())
+      .build();
+
+    SubmitResult<NfTokenMint> mintSubmitResult = xrplClient.submit(wallet, nfTokenMint);
+    assertThat(mintSubmitResult.result()).isEqualTo(TransactionResultCodes.TES_SUCCESS);
+    assertThat(mintSubmitResult.transactionResult().transaction().hash()).isNotEmpty().get()
+      .isEqualTo(mintSubmitResult.transactionResult().hash());
+
+    this.scanForResult(
+      () -> {
+        try {
+          return xrplClient.accountNfts(wallet.classicAddress());
+        } catch (JsonRpcClientErrorException e) {
+          throw new RuntimeException(e);
+        }
+      },
+      result -> result.accountNfts().size() == previousLength + 1
+    );
```

Comment:
What if we include a URI in the NFT that we mint, and check that an NFT with the same URI exists in this check